nomos-services/storage/src/api/da/mod.rs

```diff
@@ -1,9 +1,43 @@
 use std::{collections::HashSet, error::Error};
 
 use async_trait::async_trait;
+use nomos_core::da::blob::Share;
 
 pub mod requests;
 
+pub trait DaConverter<B: StorageDaApi> {
+    type Share: Share;
+    type Error: Error + Send + Sync + 'static;
+
+    fn blob_id_to_storage(
+        blob_id: <Self::Share as Share>::BlobId,
+    ) -> Result<B::BlobId, Self::Error>;
+
+    fn blob_id_from_storage(
+        blob_id: B::BlobId,
+    ) -> Result<<Self::Share as Share>::BlobId, Self::Error>;
+
+    fn share_index_to_storage(
+        share_index: <Self::Share as Share>::ShareIndex,
```

Comment:
I think one type alias for each of the different share index types would benefit a lot in this case.